packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -843,4 +846,66 @@ export class CapabilityService {
     }
     return clientsFromStripe;
   }
+
+  /**
+   * Fetch the list of capabilities for the given plan ids
+   */
+  async planIdsToClientCapabilities(
+    subscribedPrices: string[]
+  ): Promise<ClientIdCapabilityMap> {
+    const stripeCapabilities = await this.planIdsToClientCapabilitiesFromStripe(
+      subscribedPrices
+    );
+
+    if (!this.capabilityManager) {
+      if (
+        this.logToSentry(
+          'planIdsToClientCapabilities.CapabilityManagerNotFound'
+        )
+      ) {
+        Sentry.captureMessage(
+          `CapabilityManager not found.`,
+          'error' as SeverityLevel
+        );
+      }
+
+      return stripeCapabilities;
+    }
+
+    try {
+      const contentfulCapabilities =
+        await this.capabilityManager.planIdsToClientCapabilities(
+          subscribedPrices
+        );
+
+      if (
+        isEqual(
+          sortClientCapabilities(contentfulCapabilities),
+          sortClientCapabilities(stripeCapabilities)
+        )
+      ) {
+        return contentfulCapabilities;
```

Comment:
Correct me if I'm wrong, following this approach, when we deprecate the Stripe metadata, this `isEqual()` will fail correct?

So am I right in saying we'd need a follow up ticket to update the logic once we start deprecating Stripe metadata?

Reply:
@StaberindeZA Great question! And yes, I believe so, but not just for this one.

For consistency and in an effort to facilitate a more easier deprecation of Stripe metadata in future, I followed the same approach as two similar, recent PRs:

- #15855
- #15928

Since they all do the same check, it will be easier to do a search and update when the time comes. 🙂